utils/misc.ml

```diff
@@ -583,10 +583,15 @@ module Color = struct
 
   (* reasonable heuristic on whether colors should be enabled *)
   let should_enable_color () =
-    let term = try Sys.getenv "TERM" with Not_found -> "" in
-    term <> "dumb"
-    && term <> ""
-    && isatty stderr
+    try
+      let v = String.lowercase_ascii (Sys.getenv "OCAML_COLOR") in
+      List.mem v ["1";"true";"yes"]
```

Comment:
I'd say there are a few problems with this implementation:

1. You implicitly equate `""` with `Never`, I wouldn't, I would equate it with `Auto`.
2. You cannot give error messages on typos. E.g. if I write say `tru` or `y` this will be `Never`.
3. It seems to me that environment variable resolution logic is wrong. The environment variable should only be consulted if no option is specified on the command line (i.e. the environment variable changes the default value of the parameter). This function seems to be invoked below on `Auto`, so if I understand correctly in the `OCAML_COLOR=yes ocamlc -color auto` invocation I will get unconditional colors whereas it should be the `auto` behaviour (the cli takes precedence).